Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp

```diff
@@ -647,8 +647,14 @@ void GPUConnectionToWebProcess::createRemoteGPU(WebGPUIdentifier identifier, Ren
     auto* renderingBackend = it->value.get();
 
     auto addResult = m_remoteGPUMap.ensure(identifier, [&] {
-        return IPC::ScopedActiveMessageReceiveQueue { RemoteGPU::create([this] (MediaPlayerIdentifier identifier, Function<void(MediaPlayer&)>&& callback) mutable {
-            this->performWithMediaPlayerOnMainThread(identifier, WTFMove(callback));
+        return IPC::ScopedActiveMessageReceiveQueue { RemoteGPU::create([this] (std::variant<WebCore::MediaPlayerIdentifier, WebKit::RemoteVideoFrameReference> identifier, Function<void(RefPtr<WebCore::VideoFrame>)>&& callback) mutable {
+            return WTF::switchOn(identifier, [&callback, this] (WebCore::MediaPlayerIdentifier i) {
+                this->performWithMediaPlayerOnMainThread(i, [&callback](auto& player) {
+                    callback(player.videoFrameForCurrentTime());
+                });
+            }, [&callback, this] (WebKit::RemoteVideoFrameReference i) {
+                callback(this->videoFrameObjectHeap().get(RemoteVideoFrameReadReference(WTFMove(i))));
```

Comment:
if you capture by reference this is captured by default, don't need the `this->`